datautils.py

```diff
@@ -135,66 +132,81 @@ def get_c4_new(nsamples, seqlen, tokenizer):
         seqlen (int, optional): The maximum sequence length for input tokenization. Defaults to 2048.
         model_path (str, optional): The path to the pretrained model weights or full model name.
             used to detect llama to call proper tokenizer.
             see https://github.com/huggingface/transformers/issues/22222#issuecomment-1488578722 for reasons.
+        eval_mode (bool, optional). defines slice selection for 'wikitext2', 'c4', 'ptb' datasets.
+        leave False for train slice.
     Returns:
-        train_loader (torch.utils.data.DataLoader or iterable): DataLoader for the training dataset.
-        test_loader (torch.utils.data.DataLoader or iterable): DataLoader for the test dataset.
+        data (torch.utils.data.DataLoader or iterable): Data iterable for the dataset.
+    Note:
+        the popular decapoda-research Llama models have errors in tokenizer config, specifically
+        incorrect token ids for BOS, EOS. This gets corrected to ensure compatibility with transformers
+        of versions 4.29 and above.
     """
-
     set_seed(seed)
 
-    if custom_data_path:
-        dataloader = torch.load(custom_data_path)[: nsamples]
-        return dataloader, None
-
-    assert name != "custom"
-
-    if "llama" in model_path.lower():
-        tokenizer = LlamaTokenizer.from_pretrained(model_path, use_fast=False)
-
-        # fix for transformer 4.28.0.dev0 compatibility
-        if tokenizer.bos_token_id != 1 or tokenizer.eos_token_id != 2:
-            try:
-                tokenizer.bos_token_id = 1
-                tokenizer.eos_token_id = 2
-                print(f"bos/eos tokens updated: {tokenizer.bos_token_id=},  {tokenizer.eos_token_id=}")
-            except AttributeError:
-                pass
-                print(f"bos/eos tokens unchanged: {tokenizer.bos_token_id=},  {tokenizer.eos_token_id=}")
-    else:
-        tokenizer = AutoTokenizer.from_pretrained(model_path, use_fast=False)
-
-    if "wikitext2" in name:
-        return get_wikitext2(nsamples, seqlen, tokenizer)
-    elif "ptb" in name:
-        if "new" in name:
-            return get_ptb_new(nsamples, seqlen, tokenizer)
-        return get_ptb(nsamples, seqlen, tokenizer)
-    elif "c4" in name:
-        if "new" in name:
-            return get_c4_new(nsamples, seqlen, tokenizer)
-        return get_c4(nsamples, seqlen, tokenizer)
+    # for pre-tokenized datasets
+    if name.lower() == "pajama":
+        data = torch.load("./data/red_pajama_n=1024.pth")[: nsamples]
+    elif name.lower() == "refinedweb":
+        data = torch.load("./data/refined_web_n=128.pth")[: nsamples]
+    elif name.lower() == "none":
+        print("Not loading any dataset. (OK if you use no compression or methods like RTN.)")
+        return None    
+    elif os.path.isfile(name):
+        try:
+            data = torch.load(name)[:nsamples]
+        except:
```

Comment:
torch will handle this exception for us

Reply:
decided to keep it for the sake of more friendly error msg